src/s3path.jl

```diff
@@ -210,45 +228,73 @@ end
 function FilePathsBase.walkpath(fp::S3Path; kwargs...)
     # Select objects with that prefix
     objects = s3_list_objects(fp.config, fp.bucket, fp.key; delimiter="")
+    root = joinpath(fp, "/")
 
     # Construct a new Channel using a recursive internal `_walkpath!` function
-    return Channel(ctype=typeof(fp)) do chnl
-        _walkpath!(fp, fp, Iterators.Stateful(objects), chnl; kwargs...)
+    return Channel(ctype=typeof(fp), csize=128) do chnl
+        _walkpath!(root, root, Iterators.Stateful(objects), chnl; kwargs...)
     end
 end
 
 function _walkpath!(root::S3Path, prefix::S3Path, objects, chnl; topdown=true, onerror=throw, kwargs...)
+    @assert root.isdirectory
```

Comment:
If folks want I could make these argument errors, but we should never hit them during normal exection. It's mostly an extra internal check to make sure any changes to this function don't break that assumption.

Reply:
This seems fine to me